AWSTranscribeStreaming/Internal/AWSTranscribeStreamingWebSocketProviderDelegate.h

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol AWSTranscribeStreamingWebSocketProviderDelegate <NSObject>
+
+- (void)didReceiveMessage:(id)webSocket
+                  message:(id)message;
+- (void)didConnect:(id)webSocket;
+- (void)didError: (nullable id)webSocket
+           error: (NSError *)error;
+- (void)didDisconnect:(id)webSocket
+     didCloseWithCode:(NSInteger)didCloseWithCode
+               reason:(NSString *)reason
+             wasClean:(BOOL)wasClean;
+
```

Comment:
the comparison to the `WebsocketEvent` in Amplify
```
enum WebsocketEvent {
    case connect // didConnect

    case disconnect(error: Error?) // didDisconnect

    case data(WebsocketProviderResponse) // didReceiveMessage and message

    case error(Error) // didError
}
```

webSocket is an `id`, which im assuming is a String identifier for the websocket. is this really needed?

Reply:
this is the delegate not the web socket protocol. two diff things

Reply:
no it was the socket rocket class so has to be id in case its a native class or something else. id means we don't know what kind of object is being passed right?

Reply:
I fixed this to be `id<AWSTranscribeStreamingWebSocketProvider>`